src/python/pants/core/util_rules/environments.py

```diff
@@ -204,12 +204,37 @@ class RemoteExtraPlatformPropertiesField(StringSequenceField):
     )
 
 
+_NO_FALLBACK_ENVIRONMENT = "<none>"
+
+
+class RemoteFallbackEnvironmentField(StringField):
+    alias = "fallback_environment"
+    default = LOCAL_ENVIRONMENT_MATCHER
+    value: str
+    help = softwrap(
+        f"""
+        The environment to fallback to when remote execution is disabled via the global option
+        `--remote-execution`.
+
+        Must be an environment name from the option `[environments-preview].names`, the
+        special string `{LOCAL_ENVIRONMENT_MATCHER}` to use the relevant local environment, or the
+        string `{_NO_FALLBACK_ENVIRONMENT}` to error when remote execution is disabled.
```

Comment:
I think that falling back by default to the local environment is error prone, so I think that this is potentially inverted.

The chances of a local environment being compatible with the remote environment without extra consideration/attention paid to setting up all of the env vars correctly is pretty slim: and it seems better to fail fast in that case ("you probably haven't made this compatible yet"), rather than to fail slowly ("I fell back automatically for you, and now something is failing in an unexpected way mid build").

Having this be `None` (the literal value) by default, and failing if remote execution cannot be used would be preferable I think, and avoids a special case.